apps/newsletters-api/src/app/authorisation.ts

```diff
@@ -0,0 +1,42 @@
+import type { FastifyRequest } from 'fastify/types/request';
+import type { UserProfile } from '@newsletters-nx/newsletters-data-client';
+import {
+	getUserEditSchema,
+	isPartialNewsletterData,
+} from '@newsletters-nx/newsletters-data-client';
+import { permissionService } from '../services/permissions';
+
+export const hasEditAccess = async (
+	profile: UserProfile | undefined,
+): Promise<boolean> => {
+	if (!profile) return false;
+	const permissions = await permissionService.get(profile);
+	const { editTags, editNewsletters, editSignUpPage, editBraze, editOphan } =
+		permissions;
+	return [editTags, editNewsletters, editSignUpPage, editBraze, editOphan].some(
+		(permission) => permission,
+	);
+};
+
+export const isAuthorisedToUpdateNewsletter = async (
```

Comment:
Just a suggestion, but maybe this function's name should make it clear that it is checking if the user is authorised to make this partical set of updates rather than able to update generally - maybe `isAuthorisedToMakeRequestedNewsletterUpdate` ?

Reply:
Good shout